stable/elasticsearch-exporter/templates/deployment.yaml

```diff
@@ -34,7 +34,6 @@ spec:
       restartPolicy: {{ .Values.restartPolicy }}
       securityContext:
         runAsNonRoot: true
-        runAsUser: 1000
```

Comment:
Can this be a parameter? I'm quite annoyed to remove it because of non-standard Kubernetes features (like OpenShift).

Reply:
ping @svenmueller because of git blaming. :)

Reply:
FYI I also opened an issue on elasticsearch-exporter related to the runAsUser:
https://github.com/justwatchcom/elasticsearch_exporter/issues/236

Reply:
How about specifying the `securityContext` as parameter and setting the current config as default? It's added security if you can specify that the container processes are not running as root user.

Reply:
Sure, I can parametrize the whole securityContext.
Could you please first explain why runAsUser: 1000 is needed or useful?
I understand the security benefit of runAsNonRoot, but I don't see why you fix a userid.
Couldn't we just have:
```yaml
      securityContext:
        runAsNonRoot: true
```
without runAsUser ?

(OpenShift, that is stricter than a standard Kubernetes regarding security, will run the container with an arbitrary, non root, user id and require a security exception to run a container with a predefined user id),

Reply:
@bquartier good question, i also can't answer this. i agree that `runAsNonRoot` should be enough for normal cases. (the setting originates from the k8s deployment resource example https://github.com/justwatchcom/elasticsearch_exporter/blob/master/examples/kubernetes/deployment.yml#L18)

```

Reply:
@svenmueller I saw the k8s deployment example in the elasticserach_exporter. It seems as unnecessary there than it is here :). I opened an [issue](https://github.com/justwatchcom/elasticsearch_exporter/issues/236)  but could not get an answer.

I can (and likely will) move on and do as you suggest (as in the prometheus-operator chart) but would have prefer to have a clean and understandable security context.

Reply:
sorry for the late reply in the referenced issue. You can remove the run as user 1000 and replace it by `runAsNonRoot`